File: Client/lib/Models/subTransactions.dart
import 'package:intl/intl.dart';

class subTransactions {
  final String id;
  final String transactionID;
  final String sender;
  final String receiver;
  final double amount;
  final DateTime time;
  final DateTime date;
  final String type;
  final bool isCredit;

  subTransactions({
    required this.id,
    required this.transactionID,
    required this.sender,
    required this.receiver,
    required this.amount,
    required this.time,
    required this.date,
    required this.type,
    required this.isCredit
  });

  factory subTransactions.fromJson(Map<String, dynamic> json) {

    final DateFormat dateFormat = DateFormat('dd-MM-yyyy');
    final DateFormat timeFormat = DateFormat('HH:mm');

    return subTransactions(
      id: json['_id'],
      transactionID: json['transactionID'],
      sender: json['sender'],
      receiver: json['receiver'],
      amount: json['amount'],
      time: timeFormat.parse(json['time']),
      date: dateFormat.parse(json['date']),
      type: json['type'],
      isCredit: json['isCredit']
    );
  }

  Map<String, dynamic> toJson() {

    final DateFormat dateFormat = DateFormat('dd-MM-yyyy');
    final DateFormat timeFormat = DateFormat('HH:mm');

    return {
      '_id': id,
      'transactionID': transactionID,
      'sender': sender,
      'receiver': receiver,
      'amount': amount,
      'time': timeFormat.format(time),
      'date': dateFormat.format(date),
      'type': type,
      'isCredit': isCredit
    };
  }
}

File: API/Models/subTransactions.js
const mongoose = require('mongoose');

const subTransactionSchema = new mongoose.Schema({
  // id: { type: String, required: true, unique: true },
  transactionID: {type: String, ref: 'transaction', required: true},
  sender: { type: String, ref: 'User', required: true },
  receiver: { type: String, ref: 'User', required: true },
  amount: { type: Number, required: true },
  time: { type: String, default: () => new Date().toLocaleTimeString() , required: true},
  date: { type: String, default: Date.now , required: true},
  type: { type: String, required: true , default:"req"},
  isCredit: {type: Boolean, required: true, default: false}
});

const subTransactions = mongoose.model('subTransactions', subTransactionSchema);

module.exports = subTransactions;
